examples/configmapgen_controller.rs

```diff
@@ -44,15 +44,18 @@ fn object_to_owner_reference<K: Resource<DynamicType = ()>>(
 }
 
 /// Controller triggers this whenever our main object or our children changed
-async fn reconcile(generator: ConfigMapGenerator, ctx: Context<Data>) -> Result<ReconcilerAction, Error> {
+async fn reconcile(
+    generator: Arc<ConfigMapGenerator>,
+    ctx: Context<Data>,
+) -> Result<ReconcilerAction, Error> {
     log::info!("working hard");
```

Comment:
This change is the biggest breaking change here imo, and it will need a comment in the changelog to highlight it upon release

Reply:
actually don't worry about this, i will add the following to the `CHANGELOG` post merge:

## Notices

To reduce the amount of cloning done inside the runtime by the reflectors and controllers (see #786), the following change is needed on the signature of your `reconcile` functions:

```diff
-async fn reconcile(myobj: MyK, ctx: Context<Data>) -> Result<ReconcilerAction>
+async fn reconcile(myobj: Arc<MyK>, ctx: Context<Data>) -> Result<ReconcilerAction>
```